ste/sender-azureFile.go

```diff
@@ -70,102 +75,65 @@ func newAzureFilesUploader(jptm IJobPartTransferMgr, destination string, p pipel
 		return nil, err
 	}
 
-	return &azureFilesUploader{
-		jptm:       jptm,
-		fileURL:    azfile.NewFileURL(*destURL, p),
-		chunkSize:  chunkSize,
-		numChunks:  numChunks,
-		pipeline:   p,
-		pacer:      pacer,
-		md5Channel: newMd5Channel(),
+	// due to the REST parity feature added in 2019-02-02, the File APIs are no longer backward compatible
+	// so we must use the latest SDK version to stay safe
+	ctx := context.WithValue(jptm.Context(), ServiceAPIVersionOverride, azfile.ServiceVersion)
+	props, err := sip.Properties()
```

Comment:
I just realized that, technically, this is another instance of the potential perf issue we discussed in another PR, becuase this may hit the network (if the soucre is Files or S3), and these "new" methods all run on 1 goroutine.  But I'd suggest we leave that as-is for now - not least of which because the same pattern is everywhere in our senders.  I've added a backlog item for future attention on the issue, and cc'd you on that time.  I think this comment can be marke "resolved" as soon as you've read it.